src/actions/users.js

```diff
@@ -323,11 +331,11 @@ export function getMissingProfilesByIds(userIds) {
     };
 }
 
-export function getMissingProfilesByUsernames(usernames) {
-    return async (dispatch, getState) => {
+export function getMissingProfilesByUsernames(usernames: Array<string>): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
         const {profiles} = getState().entities.users;
 
-        const usernameProfiles = Object.values(profiles).reduce((acc, profile) => {
+        const usernameProfiles = Object.values(profiles).reduce((acc, profile: any) => {
```

Comment:
Can we use type `UserProfile` here instead of `any`?

Reply:
To avoid the issue: https://github.com/facebook/flow/issues/2221

Flow infers that could be `mixed`... IDK, probably something better workaround there.

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/actions/users.js:339:25

Cannot get profile.username because property username is missing in mixed [1].

     src/actions/users.js
     336│         const {profiles} = getState().entities.users;
     337│
     338│         const usernameProfiles = Object.values(profiles).reduce((acc, profile) => {
     339│             acc[profile.username] = profile;
     340│             return acc;
     341│         }, {});
     342│

     /tmp/flow/flowlib_113cee6b/core.js
 [1]  62│     static values(object: any): Array<mixed>;
```

Reply:
Cool, thanks for teaching me something!

Reply:
```suggestion
        const usernameProfiles = (Object.values(profiles): any).reduce((acc, profile: UserProfile) => {
```

Above works but maybe not necessary.